var/spack/repos/builtin/packages/pcre2/package.py

```diff
@@ -43,11 +69,17 @@ def configure_args(self):
 
         return args
 
-    @property
-    def libs(self):
-        if "+multibyte" in self.spec:
-            name = "libpcre2-32"
-        else:
-            name = "libpcre2-8"
 
-        return find_libraries(name, root=self.prefix, recursive=True)
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        args = []
+        args.append(self.define_from_variant("PCRE2_BUILD_PCRE2_16", "multibyte"))
+        args.append(self.define_from_variant("PCRE2_BUILD_PCRE2_32", "multibyte"))
+        args.append(self.define_from_variant("PCRE2_SUPPORT_JIT", "jit"))
+        args.append(self.define_from_variant("BUILD_SHARED_LIBS", "shared"))
```

Comment:
I'm not 100% sure what will happen on when building `pcre2 platform=linux` if `shared` is not an available variant there: IMO it's safest to guard this in an `if`.